src/Language/Marlowe/Runtime/Types/ContractJSON.hs

```diff
@@ -68,3 +77,36 @@ getContractJSON endpoint reqContractId = do
     let request = setRequestMethod "GET" $ setRequestHeader "Accept" ["application/json"] initialRequest
     response <- httpLBS request
     return $ eitherDecode (getResponseBody response)
+
+data Transaction = Transaction
+  { txLink :: String
+  , txBlock :: Integer
+  , txSlot :: Integer
+  , txContractId :: String
+  , txTransactionId :: String
+  } deriving (Show, Eq)
+
+instance FromJSON Transaction where
+  parseJSON = withObject "Transaction" $ \o -> do
+    link <- ((o .: "links") >>= (.: "transaction"))
+    res <- o .: "resource"
+    Transaction
+      <$> pure link
+      <*> ((res .: "block") >>= (.: "blockNo"))
+      <*> ((res .: "block") >>= (.: "slotNo"))
+      <*> res .: "contractId"
+      <*> res .: "transactionId"
```

Comment:
```suggestion
    Transaction link
            <$> ((res .: "block") >>= (.: "blockNo"))
            <*> ((res .: "block") >>= (.: "slotNo"))
            <*> res .: "contractId"
            <*> res .: "transactionId"
```
Just linting

Reply:
Fixed, actually don't even need to assign `link` at all, so I made it less code than above.